slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,378 @@
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>% 
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+Save the recipe we like
+
+<br>
+
+## Using a recipe with workflows
+
+recipes are can be used with {workflows} to "combine" it with a model
+
+<br>
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+wf_spec <- workflow() %>%  
+\ \ [add_recipe(rec_spec)]{style="color: #CA225E;"} %>%  
+\ \ add_model(linear_reg())
+:::
+
+## Recipes are estimated
+
+<br>
+
+Every preprocessing step in a recipe that involved calculations uses the *training* set. For example:
+
+- Levels of a factor
+- Determination of zero-variance
+- Normalization
+- Feature extraction
+
+Once a a recipe is added to a workflow, this occurs when `fit()` is called.
+
+
+## Debugging a recipe
+
+- Typically, you will want to use a workflow to estimate and apply a recipe.
+
+. . .
+
+- If you have an error and need to debug your recipe, the original recipe object (e.g. `rec_spec`) can be estimated manually with a function called `prep()`. It is analogous to `fit()`. See [TMwR section 16.4](https://www.tmwr.org/dimensionality.html#recipe-functions)
+
+. . .
+
+- Another function (`bake()`) is analogous to `predict()`, and gives you the processed data back.
+
+. . .
+
+- The `tidy()` function can be used to get specific results from the recipe.
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute top="0" right="0" width="150" height="150"}
+
+<br>
+
+*Take the recipe and `prep()` then `bake()` it to see what the resulting data set looks like.*
+
+*Try removing steps to see how the result changes.*
+
+<br>
+
+```{r}
+#| label: recipes-prep-exercise
+#| echo: false
+countdown(minutes = 5, id = "recipes-prep")
+```
+
+## Printing a recipe
+
+```{r}
+#| label: rec_spec
+rec_spec <- recipe(tip ~ ., data = taxi_train) %>%  
+  step_unknown(all_nominal_predictors()) %>%  
+  step_dummy(all_nominal_predictors()) %>%  
+  step_zv(all_predictors()) %>%  
+  step_log(distance) %>% 
+  step_normalize(all_numeric_predictors())
+```
+
+```{r}
+#| label: rec_spec-printing
+#| message: true
+rec_spec
+```
+
+## Prepping a recipe
+
+```{r}
+#| label: rec_spec-prep-printing
+#| message: true
+prep(rec_spec)
+```
+
+## Baking a recipe
+
+```{r}
+#| label: rec_spec-prep-bake-printing
+prep(rec_spec) %>%
+  bake(new_data = taxi_train)
+```
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute top="0" right="0" width="150" height="150"}
+
+*Take a prepped recipe and use the `tidy()` function on it.*
+
+*Use the `number` argument to inspect different steps.*
+
+<br>
+
+```{r}
+#| label: recipes-tidy-exercise
+#| echo: false
+countdown(minutes = 5, id = "recipes-tidy")
+```
+
+## Tidying a recipe
+
+```{r}
+#| label: rec_spec-tidy
+prep(rec_spec) %>%
+  tidy()
+```
+
+## Tidying a recipe
+
+```{r}
+#| label: rec_spec-tidy-2
+prep(rec_spec) %>%
+  tidy(number = 2)
+```
+
```

Comment:
Would it be worth revisiting how recipes can live inside of workflows and the `prep()`ping + `bake()`ing is handled internally then to wrap this up?

Reply:
yes, it would!

did it in [`3b798f4` (#233)](https://github.com/tidymodels/workshops/pull/233/commits/3b798f4e6031d9dc8b201a03e4a3b316b121d8b9)